code/modules/projectiles/guns/energy/special.dm

```diff
@@ -96,8 +96,66 @@
 	fire_sound = 'sound/effects/stealthoff.ogg'
 	w_class = 3.0
 	origin_tech = "combat=5;phorontech=4"
+	slot_flags = SLOT_HOLSTER | SLOT_BELT
 	projectile_type = /obj/item/projectile/energy/phoron
 
+/obj/item/weapon/gun/energy/beegun
+	name = "\improper NanoTrasen Portable Apiary"
+	desc = "An experimental firearm that converts energy into bees, for purely botanical purposes."
+	icon_state = "gyrorifle"
+	item_state = "arifle"
+	charge_meter = 0
+	w_class = 4
+	fire_sound = 'sound/effects/Buzz2.ogg'
+	force = 5
+	projectile_type = /obj/item/projectile/energy/bee
+	slot_flags = SLOT_BACK
+	max_shots = 9
+
+	firemodes = list(
+		list(name="EXTERMINATE", burst=3, burst_delay = 1, move_delay = 0, fire_delay = 0, dispersion = list(0.0, 0.2, -0.2)),
+		)
+
+/obj/item/weapon/gun/energy/mousegun
+	name = "\improper NT \"Arodentia\" Exterminator ray"
+	desc = "A highly sophisticated and certainly experimental raygun designed for rapid pest-control."
+	icon_state = "floramut100"
+	item_state = "floramut"
+	charge_meter = 0
+	w_class = 3
+	fire_sound = 'sound/weapons/taser2.ogg'
+	force = 5
+	projectile_type = /obj/item/projectile/beam/mousegun
+	slot_flags = SLOT_HOLSTER | SLOT_BELT
+	max_shots = 6
+
+	firemodes = list(
+		list(name="EXTERMINATE", burst=3, burst_delay = 1, move_delay = 0, fire_delay = 3, dispersion = list(0.0, 6,0, -6.0)),
+		)
+
+	var/lightfail = 0
+
+/obj/item/weapon/gun/energy/mousegun/handle_post_fire(mob/user, atom/target, var/pointblank=0, var/reflex=0, var/playemote = 1)
+	var/T = get_turf(user)
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+	s.set_up(3, 1, T)
+	s.start()
+	failcheck()
+	..()
+
+/obj/item/weapon/gun/energy/mousegun/proc/failcheck()
+	lightfail = 0
+	if (prob(25))
+		for (var/mob/living/M in range(rand(1,4),src)) //Big failure, TIME FOR RADIATION BITCHES
+			if (src in M.contents)
+				M << "<span class='danger'>[src]'s reactor overloads!</span>"
+			M << "<span class='warning'>You feel a wave of heat wash over you.</span>"
+			M.apply_effect(300, IRRADIATE)

```

Comment:
Not really. 30 seconds of irradiation damage does not equate to that much, although it can cause other harmful mutations. The only problem here is that the gun breaks in a one in four chance.